apps/webapp/src/components/layout.tsx

```diff
@@ -1,31 +1,40 @@
 import { LoaderFunction, Outlet, useLoaderData } from 'react-router-dom';
+import { getChainId } from '../fetchers/chain-id';
 import { HeadTag } from './metadata/head-tag';
 import { Header } from './header/header';
 import { Toaster } from '@penumbra-zone/ui';
 import '@penumbra-zone/ui/styles/globals.css';
-import { isExtensionInstalled } from '../utils/is-connected';
-import { getChainId } from '../fetchers/chain-id';
+import { ExtensionNotConnected } from './extension-not-connected';
 import { ExtensionNotInstalled } from './extension-not-installed';
 import { Footer } from './footer';
+import {
+  isPraxAvailable,
+  isPraxConnected,
+  isPraxConnectedTimeout,
+} from '@penumbra-zone/client/prax';
 
 export type LayoutLoaderResult =
-  | { isInstalled: false }
+  | { isInstalled: boolean; isConnected: boolean }
   | {
       isInstalled: true;
+      isConnected: true;
       chainId: string;
     };
 
 export const LayoutLoader: LoaderFunction = async (): Promise<LayoutLoaderResult> => {
-  const isInstalled = isExtensionInstalled();
-  if (!isInstalled) return { isInstalled };
+  const isInstalled = isPraxAvailable();
+  if (!isInstalled) return { isInstalled, isConnected: false };
+  const isConnected = isPraxConnected() || (await isPraxConnectedTimeout(1000));
```

Comment:
1 second felt a little long to me, could we get away with 500ms?

Reply:
no, we couldn't